tasks/release.py

```diff
@@ -427,12 +428,16 @@ def create_rc(ctx, release_branch, patch_version=False, upstream="origin", slack
         This also requires that there are no local uncommitted changes, that the current branch is 'main' or the
         release branch, and that no branch named 'release/<new rc version>' already exists locally or upstream.
     """
-
     major_version = get_version_major(release_branch)
 
     with agent_context(ctx, release_branch):
         github = GithubAPI(repository=GITHUB_REPO_NAME)
 
+        if github_action:
+            set_git_config('user.name', 'github-actions[bot]')
+            set_git_config('user.email', 'github-actions[bot]@users.noreply.github.com')
+            upstream = f"https://x-access-token:{os.environ.get('GITHUB_TOKEN')}@github.com/{GITHUB_REPO_NAME}.git"
```

Comment:
Are we sure this will not leak the token when executing the command?

Reply:
GitHub masks the secret by default, so the token isn't exposed ([example](https://github.com/DataDog/datadog-agent/actions/runs/12290204973/job/34296969010#step:6:750)). let me know if there are any concerns